src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,111 @@
+package mate.academy.dao;
+
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.services.ConnectionUtil;
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    @Override
+    public Book create(Book book) {
+        String query = "INSERT INTO books (title, price) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement statement = connection.prepareStatement(query,
+                     Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            int affectedRows = statement.executeUpdate();
+            checkInsertedRowsAmount(affectedRows);
+
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if(generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
```

Comment:
whink which data should be returned the one you provide in request or the one which is actually saved?

Reply:
This is not resolved. Here I wish to see that you populating book with data that actually in data base. Because you could do some manipulation with data and then save it, but in your implementation you will return not actual data